session_rpcserver.go

```diff
@@ -979,16 +1011,62 @@ func (s *sessionRpcServer) AddAutopilotSession(ctx context.Context,
 		caveats = append(caveats, firewall.MetaPrivacyCaveat)
 	}
 
+	// If a previous session ID has been set to link this new one to, we
+	// first check if we have the referenced session, and we make sure it
+	// has been revoked.
+	var prevSess *session.Session
+	if len(req.PrevLocalPub) != 0 {
+		prevPub, err := btcec.ParsePubKey(req.PrevLocalPub)
+		if err != nil {
+			return nil, err
+		}
+
+		prevSess, err = s.db.GetSession(prevPub)
+		if errors.Is(err, session.ErrSessionNotFound) {
+			return nil, fmt.Errorf("linked session(%x) not found",
+				req.PrevLocalPub)
+		} else if err != nil {
+			return nil, err
+		}
+
+		if prevSess.State != session.StateRevoked &&
+			prevSess.State != session.StateExpired {
+
+			return nil, fmt.Errorf("linked session(%x) still "+
+				"active", req.PrevLocalPub)
+		}
+	}
+
 	sess, err := session.NewSession(
 		req.Label, session.TypeAutopilot, expiry, req.MailboxServerAddr,
-		req.DevServer, perms, caveats, featureConfig, privacy,
+		req.DevServer, perms, caveats, featureConfig, privacy, prevSess,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("error creating new session: %v", err)
 	}
 
+	// If this session is being linked to a previous one, then we need to
+	// use the previous session's local private key to sign the new
+	// session's public key in order to prove to the Autopilot server that
+	// the two session's belong to the same owner.
+	var linkSig []byte
+	if prevSess != nil {
+		msg := sess.LocalPublicKey.SerializeCompressed()
+		linkSig = ecdsa.Sign(prevSess.LocalPrivateKey, msg).Serialize()
+	}
+
+	// Ensure that the session ID to group ID index is updated with the
+	// new pair before storing the session or writing any values to the
+	// privacy mapper.
+	err = s.cfg.sessionIDIndex.AddSessionAndGroupIDPair(
```

Comment:
Same comment as above here, that this might successfully store this in the index, but then fail below :).